amy/extrequests/views.py

```diff
@@ -571,53 +601,49 @@ def all_trainingrequests(request):
                     role=role,
                     defaults=dict(
-                        seat_membership=membership,
+                        seat_membership=membership_to_use,
                         seat_public=seat_public,
                         seat_open_training=open_seat,
                     ),
                 )
 
-            today = datetime.date.today()
-
+                # if membership is different for each request,
+                # collect warnings after each request is processed
+                if membership_auto_assign:
+                    warnings += [
+                        f"{r}: {w}"
+                        for w in get_membership_warnings_after_match(
+                            membership=membership_to_use,
+                            seat_public=seat_public,
+                            event=event,
+                        )
+                    ]
+
+            # if membership is the same for all matches,
+            # collect warnings after all requests are processed
             if membership:
-                remaining = (
-                    membership.public_instructor_training_seats_remaining
-                    if seat_public
-                    else membership.inhouse_instructor_training_seats_remaining
+                warnings = get_membership_warnings_after_match(
+                    membership=membership, seat_public=seat_public, event=event
                 )
-                if remaining <= 0:
-                    messages.warning(
-                        request,
-                        f'Membership "{membership}" is using more training seats than '
-                        "it's been allowed.",
-                    )
 
-                # check if membership is active
-                if not (
-                    membership.agreement_start <= today <= membership.agreement_end
-                ):
-                    messages.warning(
-                        request,
-                        f'Membership "{membership}" is not active.',
-                    )
-
-                # show warning if training falls out of agreement dates
-                if (
-                    event.start
-                    and event.start < membership.agreement_start
-                    or event.end
-                    and event.end > membership.agreement_end
-                ):
-                    messages.warning(
-                        request,
-                        f'Training "{event}" has start or end date outside '
-                        f'membership "{membership}" agreement dates.',
-                    )
-
-            messages.success(
-                request,
-                "Successfully accepted and matched selected people to training.",
-            )
+            for msg in warnings:
+                messages.warning(request, msg)
+            if errors:
+                for msg in errors:
+                    messages.error(request, msg)
```

Comment:
Nit: Can be moved outside the if-statement, ie. above L631. This way we keep similar code close to each other, and logic related to displaying the final message is separated out.